operator/api/redpanda/v1alpha3/redpanda_conversion.go

```diff
@@ -0,0 +1,954 @@
+		}
+	}
+
+	return redpandav1alpha2.Listener{
+		Enabled:              ptr.To(true),
+		Port:                 ptr.To(internal.Port),
+		AuthenticationMethod: (*string)(internal.AuthenticationMethod),
+		PrefixTemplate:       internal.PrefixTemplate,
+		TLS:                  convertListenerTLSToListenerTLS(fmt.Sprintf("%s-%s", kind, internal.Name), internal.TLS),
+	}, external
+}
+
+//nolint:gosec // unsafe is just for fancy type casting
+func convertFromListeners[T ~string](tls *redpanda.TLS, cfg redpanda.ListenerConfig[T]) []Listener {
+	listeners := make([]Listener, 1+len(cfg.External))
+
+	listeners[0] = Listener{
+		Name:                 "internal",
+		Port:                 cfg.Port,
+		AuthenticationMethod: (*AuthenticationMethod)(unsafe.Pointer(cfg.AuthenticationMethod)),
+		TLS:                  convertInternalTLSToListenerTLS(tls, cfg.TLS),
+
+		// Unsupported on the internal listener in v1alpha2.
+		AdvertisedPorts: nil,
+		PrefixTemplate:  nil,
+	}
+
+	i := 0
+	for _, name := range slices.Sorted(maps.Keys(cfg.External)) {
+		l := cfg.External[name]
+
+		if !l.IsEnabled() {
+			continue
+		}
+
+		i++
+
+		var listenerTLS *ListenerTLS
+		if l.TLS != nil && l.TLS.IsEnabled(&cfg.TLS, tls) {
+			// InternalTLS and ExternalTLS are effectively the same type sans
+			// optionality of fields. To dry this up, we just upcast external
+			// to internal.
+			listenerTLS = convertInternalTLSToListenerTLS(tls, redpanda.InternalTLS{
+				Enabled:           ptr.To(true),
+				Cert:              ptr.Deref(l.TLS.Cert, ""),
+				RequireClientAuth: ptr.Deref(l.TLS.RequireClientAuth, false),
+				TrustStore:        l.TLS.TrustStore,
+			})
+		}
+
+		listeners[i] = Listener{
+			Name:                 name,
+			Port:                 l.Port,
+			AdvertisedPorts:      l.AdvertisedPorts,
+			PrefixTemplate:       l.PrefixTemplate,
+			AuthenticationMethod: (*AuthenticationMethod)(unsafe.Pointer(l.AuthenticationMethod)),
+			TLS:                  listenerTLS,
+		}
+	}
+
+	return listeners[:i+1]
+}
+
+func convertListenerTLSToCertificate(tls *ListenerTLS) *redpandav1alpha2.Certificate {
+	if tls.IssuerRef != nil {
+		return &redpandav1alpha2.Certificate{
+			Enabled: ptr.To(true),
+			IssuerRef: &redpandav1alpha2.IssuerRef{
+				Name:  ptr.To(tls.IssuerRef.Name),
+				Kind:  (*string)(ptr.To(tls.IssuerRef.Kind)),
+				Group: nil,
+			},
+			ApplyInternalDNSNames: nil,
+			CAEnabled:             nil,
+			ClientSecretRef:       nil,
+			Duration:              nil,
+			SecretRef:             nil,
+		}
+	}
+
+	if tls.Secrets != nil {
+		var clientRef *redpandav1alpha2.SecretRef
+		if tls.Secrets.ClientSecretRef != nil {
+			clientRef = &redpandav1alpha2.SecretRef{
+				Name: ptr.To(tls.Secrets.ClientSecretRef.Name),
+			}
+		}
+
+		return &redpandav1alpha2.Certificate{
+			Enabled: ptr.To(true),
+			SecretRef: &redpandav1alpha2.SecretRef{
+				Name: ptr.To(tls.Secrets.SecretRef.Name),
+			},
+			ApplyInternalDNSNames: nil,
+			CAEnabled:             nil,
```

Comment:
deprecated? all of the above appear unused. I just worry if we haven't explicitly deprecated these fields, we'll lose behavior roundtripping.